src/components/AddBook.js

```diff
@@ -1,25 +1,70 @@
+import { useDispatch } from 'react-redux';
+import React, { useState } from 'react';
+import { addBook } from '../redux/books/books';
+
 import classes from './AddBook.module.css';
 
-const AddBook = () => (
-  <>
-    <div className={classes.formContainer}>
-      <form className={classes.form}>
-        <input
-          className={classes.inputBookTitle}
-          type="text"
-          placeholder="Book Title"
-        />
-        <select name="Economy" className={classes.catagories}>
-          <option>Catagories</option>
-          <option value="Economy">Economy</option>
-          <option value="Action">Action</option>
-          <option value="Science Fiction">Science Fiction</option>
-        </select>
-        <button className={classes.button} type="submit">
-          Add Book
-        </button>
-      </form>
-    </div>
-  </>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+
+  const titleChangeHandler = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const authorChangeHandler = (event) => {
+    setAuthor(event.target.value);
+  };
+
+  const categoryChangeHandler = (event) => {
+    setCategory(event.target.value);
+  };
+
+  const formSubmitHandler = (event) => {
+    event.preventDefault();
+    dispatch(addBook(title, author, category));
+    setAuthor('');
+    setTitle('');
+  };
```

Comment:
- [ ] This function dispatches the addBook action creator with inputs from the state. Nice job so far :+1: . However, from the requirements of this project, the function should take the inputs from the state and generate a unique id and add them to an object, then dispatches the addBook action creator. Please let's fix this issue.